_source/_posts/2022-08-15-spring-boot-crud-update.md

```diff
@@ -0,0 +1,1077 @@
+      if (!value) {
+        return
+      }
+      this.$api.createNew(value, false).then( (response) => {
+        this.$log.debug("New item created:", response)
+        this.newTodoTitle = ""
+        this.todos.push({
+          id: response.data.id,
+          title: value,
+          completed: false
+        })
+        this.$refs.newTodoInput.blur()
+      }).catch((error) => {
+        this.$log.debug(error);
+        this.error = "Failed to add todo"
+      });
+    },
+    handleCancelEditingNewTodo() {
+      this.newTodoTitle = ""
+    },
+
+    handleSetCompleted(id, value) {
+      let todo = this.todos.find(todo => id === todo.id)
+      todo.completed = value
+    },
+
+    handleSetTitle(id, value) {
+      let todo = this.todos.find(todo => id === todo.id)
+      todo.title = value
+    },
+
+    handleShowError(message) {
+      this.error = message
+    },
+
+    handleErrorClick () {
+      this.error = null;
+    },
+
+  },
+
+}
+</script>
+
+<style>
+#row-container {
+  margin-top: 100px;
+}
+.my-card {
+  min-width: 600px;
+}
+.error {
+  color: red;
+  text-align: center;
+  min-width: 600px;
+  margin-top: 10px;
+}
+</style>
+```
+{% endraw %}
+
+This component encapsulates the card that holds all of the todos, as well as the todo-associated interface elements. It also handles the rest of the functions related to updating todos on the server as well as in the local cache. 
+
+You're welcome to delete the `HelloWorld.vue` component, if you want. Or you can leave it. It's not needed.
+
+## Test the todo app
+
+Make sure the Spring Boot server is still running. In a separate Bash shell, from the resource server directory, run the following command (if it is not already still running)
+
+```bash
+./gradlew bootRun
+```
+
+Start the Vue app using the embedded development server. From the client directory:
+
+```bash
+npm run serve
+```
+
+Open a browser and navigate to [http://localhost:8080](http://localhost:8080). You'll see the "please log in" page.
+
+
+{% img blog/spring-boot-crud-update/please-log-in.png alt:"Please log in" width:"1000" %}{: .center-image }
+
+Log into the app using Okta SSO.
+
+{% img blog/spring-boot-crud-update/okta-login.png alt:"Okta SSO login" width:"600" %}{: .center-image }
+
+That will redirect you to the Todo app main screen.
+
+{% img blog/spring-boot-crud-update/app-main-screen.png alt:"Todo app main screen" width:"1000" %}{: .center-image }
+
+You should be able to delete items, add new items, rename items, and filter items. All data is stored on the Spring Boot resource server and is presented by the Vue + Quasar frontend. 
+
+## Wrapping up
+
+In this tutorial, you built a Spring Boot resource server backend and a Vue front-end. The Vue client used the latest Vue 3 version with the Quasar framework. The app included full CRUD (create, read, update, and delete) capabilities. It was all secured using Okta.
+
+If you liked this post, there's a good chance you'll like similar ones:
```

Comment:
Did you create a repo for this? What I usually do is use the existing repo from the post this one is redirected from. I created a release to store the old version, then create a PR with the new version. I merge shortly after the post is published. For an example, see https://github.com/oktadev/okta-spring-boot-saml-example/pull/17 and https://github.com/oktadev/okta-spring-boot-saml-example/releases.

Reply:
I linked to the repo in the leankit ticket.
https://github.com/moksamedia/okta-spring-boot-vue-update

Reply:
I need to update it with these changes and add a README. Will do by tomorrow morning.